### PR TITLE
add github workflow for checking openapi

### PR DESCRIPTION
**Pull request by dni**
adds workflow that validates openapi.json

relating to issue: https://github.com/lnbits/lnbits/issues/1442

### CODE REVIEW

**Comment by jackstar12** [comment]
Fix for current error:
```patch
diff --git a/lnbits/settings.py b/lnbits/settings.py
index adec5951..ae373383 100644
--- a/lnbits/settings.py
+++ b/lnbits/settings.py
@@ -228,6 +228,12 @@ class EditableSettings(
             **{k: v for k, v in d.items() if k in inspect.signature(cls).parameters}
         )
 
+    class Config:
+        @staticmethod
+        def schema_extra(schema: dict[str, Any]) -> None:
+            for prop in schema.get("properties", {}).values():
+                prop.pop("env_names", None)
+
 
 class EnvSettings(LNbitsSettings):
     debug: bool = Field(default=False)
```

**Comment by dni** [comment]
omg give this man a bounty!

File: lnbits/settings.py [hunk 1]
```diff
@@ -245,6 +245,13 @@ def from_dict(cls, d: dict):
             **{k: v for k, v in d.items() if k in inspect.signature(cls).parameters}
         )
 
+    # fixes openapi.json validation, remove field env_names
```
Comment (motorina0) [inline lnbits/settings.py:248]: - is there no side effect from removing `env_names`?
- when was `env_names` added and why?

Comment (dni) [reply lnbits/settings.py:248]: `env_names` is added by BaseSettings class https://github.com/pydantic/pydantic/blob/3703caa3725a2d9190d37f7af7aab0fa2d096fae/pydantic/v1/env_settings.py#L118

it seems to me its just additional metadata, and not uses anywhere else. i did not notice anything testing.

Comment (motorina0) [reply lnbits/settings.py:248]: got it!

**Review by motorina0 — APPROVED** [review]

**Comment by dni** [comment]
closes https://github.com/lnbits/lnbits/issues/1442

**Pull request merged**